src/main/java/com/google/devtools/build/lib/analysis/actions/CustomCommandLine.java

```diff
@@ -1297,32 +1249,33 @@ private ImmutableList<String> argumentsInternal(@Nullable ArtifactExpander artif
         arg = substituteTreeFileArtifactArgvFragment((TreeFileArtifactArgvFragment) arg);
       }
       if (arg instanceof NestedSet) {
-        evalSimpleVectorArg(((NestedSet<?>) arg).toList(), builder);
+        evalSimpleVectorArg(((NestedSet<?>) arg).toList(), builder, pathMapper);
       } else if (arg instanceof Iterable) {
-        evalSimpleVectorArg((Iterable<?>) arg, builder);
+        evalSimpleVectorArg((Iterable<?>) arg, builder, pathMapper);
       } else if (arg instanceof ArgvFragment) {
         if (artifactExpander != null && arg instanceof TreeArtifactExpansionArgvFragment) {
           TreeArtifactExpansionArgvFragment expansionArg = (TreeArtifactExpansionArgvFragment) arg;
           expansionArg.eval(builder, artifactExpander);
         } else {
-          i = ((ArgvFragment) arg).eval(arguments, i, builder, getPathStripper());
+          i = ((ArgvFragment) arg).eval(arguments, i, builder, pathMapper);
         }
       } else if (arg instanceof ActionInput) {
-        builder.add(getPathStripper().getMappedExecPathString((ActionInput) arg));
+        builder.add(pathMapper.getMappedExecPathString((ActionInput) arg));
       } else if (arg instanceof PathFragment) {
-        builder.add(getPathStripper().map((PathFragment) arg).getPathString());
+        builder.add(pathMapper.map((PathFragment) arg).getPathString());
       } else {
         builder.add(CommandLineItem.expandToCommandLine(arg));
       }
     }
     return builder.build();
   }
 
-  private void evalSimpleVectorArg(Iterable<?> arg, ImmutableList.Builder<String> builder) {
+  private void evalSimpleVectorArg(Iterable<?> arg, ImmutableList.Builder<String> builder,
+      PathMapper pathMapper) {
     for (Object value : arg) {
       builder.add(
           value instanceof ActionInput
```

Comment:
@fmeum I know you didn't change this logic here but I'm seeing another edge case where pathmapping is not working so great here because it's only handling ActionInput and nothing else. I'm seeing `CustomCommandLine.arguments` returns a list of argument objects. The argument objects could be of different types including `Iterable`. I'm seeing an Iterable that contains `String` of paths that need to be mapped. Any idea why the `ActionInput` check is here?

Reply:
also the current `StrippingPathMapper` would probably need some modification for it to work here as well. This is the format of the `String` in the `Iterable` arg I saw `someKey=blaze-out/k8-opt-exec-ST-50b1e42ff8f2/genfiles/<foo>` so the logic we have in `isOutputPath` wouldn't help in this case? 
![Screenshot 2023-09-21 at 4 45 23 PM](https://github.com/bazelbuild/bazel/assets/91614797/e3df91c3-4ee0-40f4-9124-063608b61366)

Reply:
and to be clear, this is a regression observed with this PR. I ran the same test with the previous implementation and did not see failures. I'll look into where this kind of arguments was handled previously

Reply:
Is this from a direct classpath `Turbine` action, which is public? If so, could you share more details about which argument uses this key-value style and isn't correctly affected by path mapping with this PR?

I thought that `String` stripping rather than just stripping for `ActionInput`s was something that we only did for Starlark actions, specifically at https://github.com/bazelbuild/bazel/blob/04a05677e4714434ec196c45a4ccbd8e0ef2f0ff/src/main/java/com/google/devtools/build/lib/actions/PathStripper.java#L118-L119. 

I tried to keep that logic as it was. The only place where `StringStripper` was instantiated in Bazel code was in https://github.com/bazelbuild/bazel/blob/master/src/main/java/com/google/devtools/build/lib/rules/java/JavaCompileActionBuilder.java#L330. Maybe it was used in other places internally, which isn't compatible with the approach I switched to in https://github.com/bazelbuild/bazel/pull/18098.